pkg/runtime/pubsub/subscriptions.go

```diff
@@ -243,9 +251,7 @@ func GetSubscriptionsGRPC(channel runtimev1pb.AppCallbackClient, log logger.Logg
 }
 
 // DeclarativeSelfHosted loads subscriptions from the given components path.
-func DeclarativeSelfHosted(componentsPath string, log logger.Logger) []Subscription {
-	var subs []Subscription
-
+func DeclarativeSelfHosted(componentsPath string, log logger.Logger) (subs []Subscription) {
```

Comment:
Why are the changes to this function required for the purpose of this PR?

Reply:
@yaron2  As I was going through the code to make the changes, I did a refactoring wherever appropriate

Reply:
In this case, reduced the nesting to make code a bit easier to read